### PR TITLE
DAOS-7205 DFS: use auto object class selection

### PR DESCRIPTION
**Pull request by mchaarawi**
Remove the SX default object class from DFS and generate oids using
the new API for oid generation. this means that if an oclass is not
specified to the DFS API or the DFS container creation:
- we use the container rf factor property to select the oclass for
  files and dirs.
- dirs use 1 single group for faster iteration / ls; files use MAX
  grps for better read/write BW.

Signed-off-by: Mohamad Chaarawi <mohamad.chaarawi@intel.com>

### CODE REVIEW

File: src/client/dfs/dfs.c [hunk 1]
```diff
@@ -1209,30 +1204,35 @@ dfs_cont_create(daos_handle_t poh, uuid_t co_uuid, dfs_attr_t *attr,
...
-	daos_obj_set_oid(&roots.cr_oids[0], 0, oclass, 0);
-
-	/* select oclass and generate ROOT OID */
-	rc = dfs_oclass_select(poh, dattr.da_oclass_id, &oclass);
+	rc = daos_obj_generate_oid_by_rf(poh, rf_factor, &roots.cr_oids[0],
+					 0, dattr.da_oclass_id, 0, 0);
 	if (rc) {
-		rc = daos_der2errno(rc);
-		D_GOTO(err_prop, rc);
+		D_ERROR("Failed to generate DB OID "DF_RC"\n", DP_RC(rc));
```
Comment (johannlombardi) [inline src/client/dfs/dfs.c:1223]: s/DB/SB ?

File: src/client/dfs/dfs.c [hunk 2]
```diff
@@ -1209,30 +1204,35 @@ dfs_cont_create(daos_handle_t poh, uuid_t co_uuid, dfs_attr_t *attr,
...
 	}
+
+	/* select oclass and generate ROOT OID */
 	roots.cr_oids[1].lo = RESERVED_LO;
 	roots.cr_oids[1].hi = ROOT_HI;
-	daos_obj_set_oid(&roots.cr_oids[1], 0, oclass, 0);
+	rc = daos_obj_generate_oid_by_rf(poh, rf_factor, &roots.cr_oids[1],
+					 0, dattr.da_oclass_id, 0, 0);
+	if (rc) {
+		D_ERROR("Failed to generate DB OID "DF_RC"\n", DP_RC(rc));
```
Comment (johannlombardi) [inline src/client/dfs/dfs.c:1233]: SB too?

**Review by johannlombardi — APPROVED** [review]

File: src/client/dfs/dfs.c (hunk 2 above)
Comment (mchaarawi) [reply src/client/dfs/dfs.c:1233]: this is supposed to be root.. let me fix this

**Review by johannlombardi — APPROVED** [review]

**Comment by johannlombardi** [comment]
@gnailzenh ?
Shall we try to run all master tests on this PR before landing it?

**Comment by johannlombardi** [comment]
I have retriggered the testing ...

**Comment by mchaarawi** [comment]
> @gnailzenh ?
> Shall we try to run all master tests on this PR before landing it?

running all tests seems to have problems with some snapshot and security tests that are not related. let me try daily regression ones.

**Comment by mchaarawi** [comment]
all tests passed except:
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-5387/18/tests

SecurityACL tests which @sylviachanoiyee mentioned they are failing on daily_regression?

**Review by johannlombardi — APPROVED** [review]

**Pull request merged**